scout/ecm_prep.py

```diff
@@ -8708,18 +8723,66 @@ def partition_microsegment(
+                        # in the earliest available year
+                        if isinstance(f_refr["baseline"][0]["typ_refrigerant"], dict):
+                            r_key_b_init = f_refr["baseline"][0][
+                                "typ_refrigerant"][[
+                                    y for y in f_refr["baseline"][0][
+                                        "typ_refrigerant"].keys()][0]]
+                        else:
+                            r_key_b_init = f_refr["baseline"][0]["typ_refrigerant"]
+                        # Use key above to pull installed base refrigerant GWP
+                        try:
+                            base_gwp_yr_init = self.handyvars.fug_emissions[
+                                "refrigerants"]["refrigerant_GWP100"][r_key_b_init]
+                        except KeyError:
+                            raise ValueError(
+                                "Refrigerant '" + r_key_b_init +
+                                "' has no available GWP data in supporting "
+                                "file ./supporting_data/convert_data/"
+                                "fugitive_emissions_convert.json")
+
+                    # Case where user assumes typical refrigerants as of
+                    # earliest year in the refrigerants data do not phase out
+                    # in subsequent years
+                    if opts.fugitive_emissions[1] == '1':
+                        # Find key to use in pulling global warming potential
+                        # value for the typical baseline tech. refrigerant
+                        # in the earliest available year
+                        if isinstance(f_refr["baseline"][0]["typ_refrigerant"], dict):
+                            r_key_b = f_refr["baseline"][0][
+                                "typ_refrigerant"][[
+                                    y for y in f_refr["baseline"][0][
+                                        "typ_refrigerant"].keys()][0]]
+                        else:
+                            r_key_b = f_refr["baseline"][0]["typ_refrigerant"]
+                    # Case where user assumes typical refrigerants have phase-
+                    # out years
+                    elif opts.fugitive_emissions[1] in ['2', '3']:
+                        # Find key to use in pulling global warming potential
+                        # value for the typical baseline tech. refrigerant;
+                        # typical refrigerants may be stored as a dict keyed in
+                        # by year
+                        if isinstance(f_refr["baseline"][0]["typ_refrigerant"], dict):
+                            r_key_b = f_refr["baseline"][0][
+                                "typ_refrigerant"][
+                                [y for y in f_refr["baseline"][0][
+                                    "typ_refrigerant"].keys() if
+                                    int(yr) >= int(y)][-1]]
```

Comment:
Possibly. Do you have any thoughts on a possible fix?

Reply:
Maybe something like:

```
if isinstance(typ_refrig, dict):
    prev_years = [int(y) for y in typ_refrig.keys() if int(yr) >= int(y)]
    if prev_years:
        r_key_b = typ_refrig[max(prev_years)]
    else:
        raise ValueError(".... ")
```

There are a number of similar sections of code, it may make sense to check for isinstance(typ_refrig, dict) and define prev_years when typ_refrig gets defined.

Reply:
This modification is implemented in d4b6094.